File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []
    day = deque()
    for x,y in zip(progresses,speeds):
        if (100-x)%y == 0:
            day.append((100-x)//y)
        else:
            day.append((100-x)//y+1)
    tmp = []
    while day:
        inx = day.popleft()
        if len(tmp) == 0:
            tmp.append(inx)
        else:
            if inx > tmp[0]:
                print(inx,tmp[0])
                answer.append(len(tmp))
                tmp = []
                tmp.append(inx)
            else:
                tmp.append(inx)  
    answer.append(len(tmp))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    for(int i=0;i<speeds.size();i++){
        int cnt = 0;
        while(progresses[i]<100){
            cnt++;
            progresses[i]+=speeds[i];
        }
        q.push(cnt);
    }
    int cnt2 = 0;
    int max = 0;
    while (!q.empty()) {
        int a = q.front();
        cnt2++;
        q.pop();
        if(a>max){
            max = a;
        }
        if(max<q.front()){
            answer.push_back(cnt2);
            cnt2 =0;
        } else if(q.empty()){
            answer.push_back(cnt2);
        }
        

	}
    return answer;
}
